### PR TITLE
feat(tools): config file

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did
Adds an optional unified config file for custom elements manifest, dev server, and docs pages

Create a `.pfe.config.json` file at the root of your project to customize the pfe tools builds.

The default values are:

```json
{
  "tagPrefix": "pfe",
  "demoURLPrefix": "https://patternflyelements.org/",
  "sourceControlURLPrefix": "https://github.com/patternfly/patternfly-elements/tree/main/",
  "aliases": {},
  "site": {
    "title": "PatternFly Elements",
    "description": "PatternFly Elements: A set of community-created web components based on PatternFly design.",
    "favicon": "/brand/logo/svg/pfe-icon-blue.svg",
    "logoUrl": "/brand/logo/svg/pfe-icon-white-shaded.svg",
    "stylesheets": []
  }
}
```

See `@patternfly/pfe-tools/config.ts` for more information.

This file is consulted when generating
- custom elements manifests
- the dev server ui
- docs pages

And this config will let us also easily specify aliases for components e.g. `rh-cta` => `Call to Action`

## YTHO

see https://github.com/RedHat-UX/red-hat-design-system/pull/495

### CODE REVIEW

**Review by zeroedin — COMMENTED** [review]
This looks good to me, was able to easily add aliases changing CTA link text to Call to action.  Nice improvement and will be helpful upstream.

**Comment by brianferry** [comment]
Same as @zeroedin , looks good to me.  Do we think we should add a README file for this so we can document the options available & how to use them?

**Comment by bennypowers** [comment]
https://github.com/RedHat-UX/red-hat-design-system/pull/495#issuecomment-1226783544 should be fixed now

**Comment by bennypowers** [comment]
@brianferry thanks for the suggestion, I added some docs in README

**Comment by brianferry** [comment]
> @brianferry thanks for the suggestion, I added some docs in README

Great, looks good to me, tested with aliases on pfe & rhds and those looked to be working.  Just to be clear @bennypowers these are not going to be changing the links on the docs pages like when running `npm run watch:docs` but will when running `npm run start` or should this work in both scenarios?

**Comment by bennypowers** [comment]
> 

do you mean that edits to the config file don't reflect on the docs site until you kill the 11ty watcher?

I'd want to fix that but dont want to block on it, can you open a new issue with a repro, after we merge?

**Review by brianferry — APPROVED** [review]

**Pull request merged**
